src/pages/dashboard/components/Chart.tsx

```diff
@@ -58,13 +55,31 @@ export function Chart(props: Props) {
     return dates;
   };
 
+  const getRecordIndex = (data: LineChartData | undefined, date: string) => {
+    let recordIndex = -1;
+
+    if (!data || !date) return recordIndex;
+
+    data.forEach((entry, index) => {
+      if (entry.date === date) {
+        recordIndex = index;
+      } else if (index + 1 <= data.length) {
+        if (
+          data[index] &&
+          data[index + 1] &&
+          new Date(date) >= new Date(data[index].date) &&
+          new Date(date) <= new Date(data[index + 1].date)
+        ) {
+          recordIndex = index;
+        }
+      }
+    });
+
+    return recordIndex;
+  };
```

Comment:
This looks really complex. Can we assign internal IDs and then look them up like that? Perhaps .find on records?

Reply:
@beganovich I'm not sure if it's possible to handle IDs in this situation. The problem arises when dealing with week/month intervals. In these cases, we need to compare dates. When a user selects a month, there are only a few dates available, making it difficult to accurately match them and display the correct chart. Comparing the dates is the only practical solution. I've simplified the approach, and I hope you agree with it.